### PR TITLE
Update CHANGELOG format and parsing

### PR DESCRIPTION
**Pull request by JakeQuilty**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -46,7 +46,13 @@ pipeline {
...
       }
     }
-
+    stage('Validate') {
+      parallel {
+        stage('Changelog') {
+          steps { sh './parse-changelog.sh' }
+        }
+      }
+    }
```
Comment (sgnn7) [inline Jenkinsfile]: nit: Newline needed after this line to match other stages

File: parse-changelog.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+cd "$(dirname "$0")"
+
+docker run --rm \
+  -v "$PWD/.:/work" \
+  -w "/work" \
+  ruby:2.5 bash -ec "
+    gem install -N parse_a_changelog
+    parse ./CHANGELOG.md
+  "
+
```
Comment (sgnn7) [inline parse-changelog.sh]: nit: Stray newline

File: parse-changelog.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -ex
+
+cd "$(dirname "$0")"
+
+docker run --rm \
+  -v "$PWD/.:/work" \
```
Comment (diverdane) [inline parse-changelog.sh]: I think this can be:
```
  -v "$PWD:/work" \
```

**Comment by JakeQuilty** [comment]
I'm not sure why this is failing with this error in Jenkins:
 ```
[2020-02-20T17:13:28.226Z] + ./bin/parse-changelog.sh
[2020-02-20T17:13:28.226Z] /var/lib/jenkins/workspace/_-summon_update-changelog-format@tmp/durable-eeacee8a/script.sh: line 1: ./bin/parse-changelog.sh: No such file or directory
script returned exit code 1
```
When I run the command locally it finds the file and passes. I checked spelling too.

File: CHANGELOG.md [hunk 4]
```diff
@@ -39,111 +37,141 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
-## [v0.1.1](https://github.com/cyberark/summon/releases/tag/v0.1.1) - 2015-06-09
+## [0.1.1] - 2015-06-09
 ### Changed
-* Attach stdin to allow running interactive processes wrapped with summon
-* Changed name from 'cauldron' to 'summon'
+- Attach stdin to allow running interactive processes wrapped with summon
+- Changed name from 'cauldron' to 'summon'
 
-## [v0.1.0](https://github.com/cyberark/summon/releases/tag/v0.1.0) - 2015-06-03
+## 0.1.0 - 2015-06-03
```
Comment (diverdane) [inline CHANGELOG.md]: Should be able to use hyperlink to [0.1.0] (initial release)

Comment (sgnn7) [reply CHANGELOG.md]: Fixed

**Pull request merged**